shared-lib/lib-ot/src/core/document/position.rs

```diff
@@ -0,0 +1,46 @@
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Position(pub Vec<usize>);
+
+impl Position {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+}
+
+impl Position {
+    // delta is default to be 1
+    pub fn transform(pre_insert_path: &Position, b: &Position, delta: i64) -> Position {
```

Comment:
Would it be better to replace delta with offset? Because we have a struct named delta.